cnxepub/templates/exercise_template.xhtml.jinja

```diff
@@ -0,0 +1,50 @@
+<div 
```

Comment:
:thinking: I noticed that some of the data-type attributes contained their parent (`exercise-stimulus`) and others did not (`question`, `option`) and that got me thinking.

I'm not sure if there was a reason behind one vs another but for greppability I like the hyphenated types. It also helps with validation to distinguish between the in-CNXML exercises and the injected ones because the markup is very different. Would something like this work?

```html
<div 
    data-type="injected-exercise"
    data-injected-from-nickname="{{ data.nickname }}"
    data-injected-from-version="{{ data.version }}"
    data-injected-from-url="{{ data.url }}"
    data-tags="{{ ' '.join(data.tags) }}"
    data-is-vocab="{{ data.is_vocab }}">
    {% if data.required_context -%}
    <div data-type="exercise-context"
        data-context-module="{{ data.required_context.module }}"
        data-context-feature="{{ data.required_context.feature }}">
        <a class="autogenerated-content"
        href="#{{ data.required_context.ref }}">[link]</a>
    </div>
    {% endif -%}
    {% if data.stimulus_html -%}
    <div data-type="exercise-stimulus">{{ data.stimulus_html }}</div>
    {% endif -%}
    {% for question in data.questions -%}
    <div 
        data-type="exercise-question"
        data-is-answer-order-important="{{ question.is_answer_order_important }}"
        data-formats="{{ ' '.join(question.formats) }}">
        {% if question.stimulus_html -%}
        <div data-type="question-stimulus">{{ question.stimulus_html }}</div>
        {% endif -%}
        <div data-type="question-stem">{{ question.stem_html }}</div>
        {% if question.answers -%}
        <ol data-type="question-answers" type="a">
            {% for answer in question.answers -%}
            <li data-type="question-answer" data-id="{{ answer.id }}" data-correctness="{{ answer.correctness }}">
                <div data-type="answer-content">{{ answer.content_html }}</div>
                {% if answer.feedback_html %}<div data-type="answer-feedback">{{ option.feedback_html }}</div>{% endif %}
            </li>
            {% endfor -%}
        </ol>
        {% endif -%}
        {% for solution in question.collaborator_solutions -%}
        <div data-type="question-solution" data-solution-source="collaborator" data-solution-type="{{ solution.solution_type }}">
            {{ solution.content_html }}
        </div>
        {% endfor -%}
        {% for solution in question.community_solutions -%}
        <div data-type="question-solution" data-solution-source="community" data-solution-type="{{ solution.solution_type }}">
            {{ solution.content_html }}
        </div>
        {% endfor -%}
    </div>
    {% endfor -%}
</div>
```